src/language.rs

```diff
@@ -95,81 +32,79 @@ pub static ALL_SUPPORTED_LANGUAGES: BySupportedLanguage<SupportedLanguage> = {
-    [
-        tree_sitter_rust::language(),
-        tree_sitter_typescript::language_tsx(),
-        tree_sitter_javascript::language(),
-        tree_sitter_swift::language(),
-        tree_sitter_objc::language(),
-        tree_sitter_toml::language(),
-        tree_sitter_python::language(),
-        tree_sitter_ruby::language(),
-        tree_sitter_c::language(),
-        tree_sitter_cpp::language(),
-        tree_sitter_go::language(),
-        tree_sitter_java::language(),
-        tree_sitter_c_sharp::language(),
-        tree_sitter_kotlin::language(),
-        tree_sitter_elisp::language(),
-        tree_sitter_elm::language(),
-        tree_sitter_dockerfile::language(),
-        tree_sitter_html::language(),
-        tree_sitter_query::language(),
-        tree_sitter_json::language(),
-        tree_sitter_css::language(),
-        tree_sitter_lua::language(),
-    ]
-});
+impl SupportedLanguage {
+    pub fn language(&self) -> Language {
+        SUPPORTED_LANGUAGE_LANGUAGES[*self]
+    }
 
-#[inline(always)]
-pub fn get_supported_language_language(supported_language: SupportedLanguage) -> Language {
-    SUPPORTED_LANGUAGE_LANGUAGES[supported_language]
+    pub fn name_for_ignore_select(&self) -> &'static str {
+        SUPPORTED_LANGUAGE_NAMES_FOR_IGNORE_SELECT[*self]
+    }
 }
 
-pub static SUPPORTED_LANGUAGE_NAMES_FOR_IGNORE_SELECT: BySupportedLanguage<&'static str> = [
-    "rust",
-    "ts",
-    "js",
-    "swift",
-    "objc",
-    "toml",
-    "py",
-    "ruby",
-    "c",
-    "cpp",
-    "go",
-    "java",
-    "csharp",
-    "kotlin",
-    "elisp",
-    "elm",
-    "docker",
-    "html",
-    "treesitterquery",
-    "json",
-    "css",
-    "lua",
-];
+static SUPPORTED_LANGUAGE_LANGUAGES: Lazy<BySupportedLanguage<Language>> = Lazy::new(|| {
+    by_supported_language!(
+        Rust => tree_sitter_rust::language(),
+        Typescript => tree_sitter_typescript::language_tsx(),
+        Javascript => tree_sitter_javascript::language(),
+        Swift => tree_sitter_swift::language(),
+        ObjectiveC => tree_sitter_objc::language(),
+        Toml => tree_sitter_toml::language(),
+        Python => tree_sitter_python::language(),
+        Ruby => tree_sitter_ruby::language(),
+        C => tree_sitter_c::language(),
+        Cpp => tree_sitter_cpp::language(),
+        Go => tree_sitter_go::language(),
+        Java => tree_sitter_java::language(),
+        CSharp => tree_sitter_c_sharp::language(),
+        Kotlin => tree_sitter_kotlin::language(),
+        Elisp => tree_sitter_elisp::language(),
+        Elm => tree_sitter_elm::language(),
+        Dockerfile => tree_sitter_dockerfile::language(),
+        Html => tree_sitter_html::language(),
+        TreeSitterQuery => tree_sitter_query::language(),
+        Json => tree_sitter_json::language(),
+        Css => tree_sitter_css::language(),
+        Lua => tree_sitter_lua::language(),
```

Comment:
How would you feature-gate these (if you wanted to eventually), since you can't use the normal annotation in a macro. (Although I suppose you could make the macro understand that annotation?)

Reply:
> Although I suppose you could make the macro understand that annotation?

Ya per I think a comment on a subsequent PR (where I did `cpp` -> `c++`), if I parse into a baked-in `syn` type eg `syn::ExprArray` it parses annotations for me

But... this isn't real Rust syntax so couldn't use that directly

Maybe that would be an excuse to mimic how eg `syn` implements `.parse::<ExprArray>()` but have the "item" parsing be different or whatever